File: trees/crearNodo/crearNodo.pas
// Definimos un procedimiento llamado "crearNodo" que toma un puntero a un nodo de árbol (de tipo "arbol") y un valor entero como argumentos
procedure crearNodo (var nodo: arbol; valor: integer);
begin
    // Creamos un nuevo nodo de árbol y asignamos su dirección al puntero "nodo"
    new(nodo);
    // Asignamos el valor proporcionado al campo "valor" del nodo
    nodo^.valor := valor;
    // Inicializamos los punteros "izquierdo" y "derecho" del nodo como nulos
    nodo^.izquierdo := nil;
    nodo^.derecho := nil;
end;

File: trees/crearNodo/crearNodo.ts
// Importamos el tipo Arbol desde su archivo correspondiente
import { Arbol } from "../type"

// Definimos una función llamada "crearNodo" que toma un valor numérico como argumento
export const crearNodo = (valor: number): Arbol => {
  // Creamos un nuevo nodo de árbol con el valor proporcionado
  const nodo = new Arbol(valor)
  // Devolvemos el nodo creado
  return nodo
}
